### ISSUE TITLE
Improve S2432 by ignoring return statements in nested functions

### ISSUE THREAD

**Issue by AMorgaut** [SonarSource/SonarJS#1355]
**I want to report a bug.**

**SonarJS version:** 5.2.1<br>
**SonarQube version:** Community Edition Version 7.8 (build 26217)<br>
**SonarLint version:** 4.1.1.3345<br>

**Rule key:** RSPEC-2432

**Reproducer** 
```js
Bar.prototype = {
    set foo(value) {
        const callback = (result) => {
            result.push(value);
            return result;  // False-positive here
        };
        report('bar', callback);
    }
};
```

**Expected behavior**

Should not raise an error

**Comment by AMorgaut** [comment]
Another situation in which it can happen
```
foo = {
    set promise(promise) {
        this._promise = promise
            .then((result) => {
                this.updateExpirationTime();
                return result;  // False-positive here
            })
            .catch((error) => {
                this._promise = null;
                return Promise.reject(error);  // False-positive here
            });
    }
};

```

**Comment by yassin-kammoun-sonarsource** [comment]
These false positive were fixed with #2377.
